client/reader/stream/index.jsx

```diff
@@ -325,60 +255,74 @@ class ReaderStream extends React.Component {
+					feedId: candidateItem.feedId,
+					blogId: candidateItem.blogId,
+				};
+				this.props.selectItem( { streamKey, postKey } );
 			}
 
-			// find the index of the post / gap in the posts array.
+			// find the index of the post / gap in the items array.
 			// Start the search from the index in the items array, which has to be equal to or larger than
-			// the index in the posts array.
+			// the index in the items array.
 			// Use lastIndexOf to walk the array from right to left
-			const selectedPostKey = findLast( posts, items[ index ], index );
-			if ( keysAreEqual( selectedPostKey, this.state.selectedPostKey ) ) {
-				selectNextItem( this.props.postsStore.id );
+			const selectedPostKey = findLast( items, items[ index ], index );
+			if ( keysAreEqual( selectedPostKey, this.props.selectedPostKey ) ) {
+				this.props.selectNextItem( { streamKey, items } );
 			} else {
-				selectItem( this.props.postsStore.id, selectedPostKey );
+				this.props.selectItem( { streamKey, postKey: selectedPostKey } );
 			}
 		}
 	};
 
 	selectPrevItem = () => {
+		// note that we grab the items directly from the stream because we don't want the transformed
+		// one with combined cards
+		const { streamKey, selectedPostKey, stream: { items } } = this.props;
 		// unlike selectNextItem, we don't want any magic here. Just move back an item if the user
 		// currently has a selected item. Otherwise do nothing.
 		// We avoid the magic here because we expect users to enter the flow using next, not previous.
-		if ( this.state.selectedPostKey ) {
-			selectPrevItem( this.props.postsStore.id );
+		if ( selectedPostKey ) {
+			this.props.selectPrevItem( { streamKey, items } );
 		}
 	};
 
+	poll = () => {
+		const { streamKey } = this.props;
+		this.props.requestPage( { streamKey, isPoll: true } );
+	};
+
 	fetchNextPage = options => {
-		if ( this.props.postsStore.isLastPage() || this.props.postsStore.isFetchingNextPage() ) {
-			return;
-		}
+		const { streamKey, stream, startDate } = this.props;
 		if ( options.triggeredByScroll ) {
-			this.props.trackScrollPage( this.props.postsStore.getPage() + 1 );
+			this.props.trackScrollPage( this.page );
+			this.page++;
 		}
-		fetchNextPage( this.props.postsStore.id );
+		const indexOfColon = streamKey.indexOf( ':' );
+		const streamType = indexOfColon === -1 ? streamKey : streamKey.substring( 0, indexOfColon );
+		const pageHandle = includes( streamType, 'rec' ) //|| streamType === 'search' // elasticsearch requires offsets
+			? { offset: stream.items.length + 1 }
+			: stream.pageHandle || { before: startDate };
+		this.props.requestPage( { streamKey, pageHandle } );
 	};
 
 	showUpdates = () => {
+		const { streamKey } = this.props;
 		this.props.onUpdatesShown();
-		showUpdates( this.props.postsStore.id );
-		if ( this.props.recommendationsStore ) {
-			shufflePosts( this.props.recommendationsStore.id );
-		}
+		this.props.showUpdates( { streamKey } );
+		// @todo: do we need to shuffle?
```

Comment:
the shuffle was to prevent us from showing the same recs in the same relative order after a page "refresh"... unsure if we really need it.

Reply:
depends on what you mean by page refresh. If the browser is actually refreshed then they will see brand new recs due to a new seed.